### PR TITLE
Improved API v2 serializer fields

### PR DESCRIPTION
**Pull request by rowanseymour**
Introduces a bunch of new custom field classes which simplify the v2 serializers, and provide consistent validation and error feedback

All deserialized lists are now limited to 100 items

### CODE REVIEW

File: temba/contacts/tests.py [hunk 1]
```diff
@@ -493,7 +493,7 @@ def setUp(self):
         self.manager1 = self.create_user("mike")
 
         self.joe = self.create_contact(name="Joe Blow", number="123", twitter="blow80")
-        self.frank = self.create_contact(name="Frank Smith", number="1234")
+        self.frank = self.create_contact(name="Frank Smith", number="12345")
```
Comment (rowanseymour) [inline temba/contacts/tests.py]: This is to fix a sorta unrelated test error where 1234 collides with the pk of the contact

File: temba/orgs/models.py [hunk 2]
```diff
@@ -835,6 +835,9 @@ def get_country_code(self):
         """
         Gets the 2-digit country code, e.g. RW, US
         """
+        return get_cacheable_attr(self, '_country_code', lambda: self.calculate_country_code())
```
Comment (rowanseymour) [inline temba/orgs/models.py]: If we have a bunch of URNs to validate this avoids fetching org channels repeatedly to calculate the country code

File: temba/channels/tests.py [hunk 3]
```diff
@@ -1088,6 +1088,8 @@ def test_register_and_claim_android(self):
         # a different country altogether should just give us the default
         us_channel = self.org.get_send_channel(TEL_SCHEME, ContactURN.create(self.org, None, 'tel:+593997290044'))
         self.assertEqual(us_channel, channel)
+
+        self.org = Org.objects.get(pk=self.org.pk)
```
Comment (rowanseymour) [inline temba/channels/tests.py]: Country code is now cached (see below)

File: temba/api/v2/fields.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            raise serializers.ValidationError("Invalid URN: %s" % data)
+
+        return normalized
+
+
+class URNListField(LimitedListField):
+    child = URNField()
+
+
+class TembaModelField(serializers.RelatedField):
```
Comment (nicpottier) [inline temba/api/v2/fields.py]: neato!

File: temba/channels/tests.py (hunk 3 above)
Comment (nicpottier) [inline temba/channels/tests.py]: Thought we were doing `id` not `pk`?

**Comment by nicpottier** [comment]
way better, nice! 🎲

**Pull request merged**